_posts/2019-08-09-iglu-scala-client-0.6.0-released/index.md

```diff
@@ -0,0 +1,116 @@
+
+The Client class defines only one function:
+
+{% highlight scala %}
+def check[F[_]: RegistryLookup: Clock: Monad, A](instance: SelfDescribingData[A]): EitherT[F, ClientError, Unit]
+{% endhighlight %}
+
+Where:
+
+* F[_] is an abstract effect type, requiring a [tagless final][tagless-final] capabilities RegistryLookup and `cats.effect.Clock` as well as instance of cats.Monad type class. Two most common concrete effect types are cats.effect.IO and Id, all necessary machinery for them provided out-of-box, but user can also define this machinery for ZIO, Monix Task or sophisticated test type
+* A is a type of self-describing entity, such as JSON. Since 0.6.0, Iglu Client is based primarily on [circe][circe] Json type, but we're trying to leave it generic whenever possible
+* ClientError is a possible unsuccessful outcome, either at resolution or validation step. Unlike ProcessingMessage from pre-0.6.0 it provides type-safe and well-structured information about the failure. This type is widely used in upcoming Snowplow bad rows
+
+From this very short excerpt an astute Scala developer might notice that we replaced several libraries with their modern counterparts:  
+
+* [Circe][circe] is used instead of [Json4s][json4s]. Circe provides much better performance characteristics, does not rely on runtime reflection, provides very clean idiomatic API and remains one of the most popular JSON libraries in Scala ecosystem for last couple of years
+* [Cats Effect][cats-effect] is used for managing side-effects, instead of implicit effect management. One can use Iglu Client 0.6.0 without bothering about Cats Effect, but it is highly recommended in async-heavy environments
+* [Cats][cats] is used instead of [Scalaz][scalaz] as FP library of choice. Cats is a transitive dependency of Circe and Cats Effect, which makes it a natural choice, since no other dependencies are shipped with Scalaz.
+* [networknt json-schema-validator][networknt-validator] is used instead of FGE JSON Validator. One more change library, driving our bad rows effort.  networknt json-schema-validator is actively maintained, provides clean API and shows very impressive results in benchmarks
+
+You can find more usage examples on dedicated [wiki page][iglu-client-docs].
+
+<h3 id="semantic-changes">2. Semantic Changes</h3>
+
+In a batch ETL world, we tried to reduce the load on Iglu Registries by leveraging very simple retry-and-cache algorithm, that was making some configurable amount of attempts before deciding that schema is missing or invalid and caching this failure. The only thing that potentially could reset this cached value is cacheTtl property, that would force the resolver to retry no matter whether cached value is success (in case somebody mutated schema) or failure (in case the registry had a long outage).
```

Comment:
In a batch ETL world, we tried to reduce the load on Iglu Registries by leveraging a very simple retry-and-cache algorithm that was making some configurable attempts before deciding whether the schema is missing or invalid and caching this failure. The only thing that potentially could reset this cached value is the cacheTtl property, that would force the resolver to retry whether the cached value was a success (in case somebody mutated schema) or a failure (in case the registry had a long outage).